### ISSUE TITLE
Build docs offline

### ISSUE THREAD

**Issue by dschwoerer** [git-cola/git-cola#1250]
Is there a way to build the documentation offline?

For Fedora building is done offline. Before 4.0 I could build docs offline, now this is not working anymore.

It fails with:

```
make -C docs all
make[1]: Entering directory '/builddir/build/BUILD/git-cola-4.0.1/docs'
/usr/bin/sphinx-build-3.11 -b html -d _build/doctrees  -a . _build/html
Running Sphinx v5.0.2
making output directory... done
WARNING: Retrying (Retry(total=4, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7fdc6ad72a10>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/setuptools/
WARNING: Retrying (Retry(total=3, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7fdc6ad71c10>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/setuptools/
WARNING: Retrying (Retry(total=2, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7fdc6ad91dd0>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/setuptools/
WARNING: Retrying (Retry(total=1, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7fdc6ad92810>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/setuptools/
WARNING: Retrying (Retry(total=0, connect=None, read=None, redirect=None, status=None)) after connection broken by 'NewConnectionError('<pip._vendor.urllib3.connection.HTTPSConnection object at 0x7fdc6ad93390>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution')': /simple/setuptools/
ERROR: Could not find a version that satisfies the requirement setuptools>=42 (from versions: none)
ERROR: No matching distribution found for setuptools>=42
WARNING: There was an error checking the latest version of pip.

Extension error (jaraco.packaging.sphinx):
Handler <function load_config_from_setup at 0x7febe0378fe0> for event 'builder-inited' threw an exception (exception: Command '['/usr/bin/python3', '-m', 'pip', 'install', '--ignore-installed', '--prefix', '/tmp/pep517-build-env-r54xa81z', 'setuptools>=42', 'setuptools_scm[toml]>=3.4.1']' returned non-zero exit status 1.)
make[1]: Leaving directory '/builddir/build/BUILD/git-cola-4.0.1/docs'
make[1]: *** [Makefile:51: html] Error 2
make: *** [Makefile:128: doc] Error 2
```

**Comment by davvid** [comment]
That's a bummer. I wonder.. is this because `pip` is trying to install the doc dependencies (perhaps via `jaraco.packaging.sphinx`).

This issue seems related:
https://github.com/jaraco/jaraco.packaging/issues/7

For the record -- I started wiring in jaraco.packaging.sphinx for the documentation (so that we can leverage rst.linker and such) but we do not technically depend on it currently.

In other words, I think we could rip those parts out and stuff should still work fine as far as doc-building is concerned. I did want to start using `rst.linker` at some point, but it's pretty low priority and having offline docs is more important IMO.

If you purge the `jaraco.packaging.sphinx` references, do it build offline okay?

I recently changed the build system to be PEP-517/518 compatible. This is all kinda a fallout of that stuff. Sorry for making the packaging harder.

This is the first time we've changed the build system in over 10+ years so I promise not to change it again anytime soon ;-)
